src/pyhf/infer/__init__.py

```diff
@@ -118,15 +126,31 @@ def hypotest(
     """
     init_pars = init_pars or pdf.config.suggested_init()
     par_bounds = par_bounds or pdf.config.suggested_bounds()
-    tensorlib, _ = get_backend()
-
-    calc = AsymptoticCalculator(data, pdf, init_pars, par_bounds, qtilde=qtilde)
+    # get fixed vals from the model
+    model_fixed_vals = [
+        (index, init)
+        for index, (init, is_fixed) in enumerate(
+            zip(init_pars, pdf.config.suggested_fixed())
+        )
+        if is_fixed
+    ]
+    # add user-defined ones at the end
+    fixed_vals = model_fixed_vals + (fixed_vals or [])
+
+    # de-dupe and use last-appended result for each index
+    fixed_vals = list(dict(fixed_vals).items())
```

Comment:
@kratsg can you add the simple example you showed me of this to the PR body for posterity?

```python
>>> foo = [('a', 1), ('a', 2)]
>>> list(dict(foo))
[('a', 2)]
```